### PR TITLE
Chore: Add type hints to attribute definitions

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Small enhancement in attribute definitions.

## Additional info
I got lost a little when I was developing something related attribute definitons lately, during that time I've added typehints so I don't have to go back and forth. The original PR is merged so not I though I'll create PR adding the typehints.

## Testing notes:
1. Make sure it makes sense.

Resolves https://github.com/ynput/ayon-core/issues/993

### CODE REVIEW

**Review by kalisp — APPROVED** [review]
LGTM

File: client/ayon_core/lib/attribute_definitions.py [hunk 1]
```diff
@@ -6,82 +6,53 @@
...
-from typing import Any, Optional
+import typing
+from typing import (
+    Any,
+    Optional,
+    List,
+    Set,
+    Dict,
+    Iterable,
+    TypedDict,
```
Comment (BigRoy) [inline client/ayon_core/lib/attribute_definitions.py:17]: TypedDict is Py3.8+ so adding this drops support for Py3.7

Comment (iLLiCiTiT) [reply client/ayon_core/lib/attribute_definitions.py:17]: Do we support 3.7 now?

**Comment by BigRoy** [comment]
Just confirming that it works in Maya 2022.3 with Py3.7.7

**Pull request merged**